src/app/components/SelectedWallet.tsx

```diff
@@ -19,43 +19,18 @@ export const SelectedWallet = () => {
     selectedWallet,
     selectedAccount,
     disconnectWallet,
-    getAccountBalance,
     chainId,
-    processErrorMessage,
+    currentBalance,
   } = useWalletProvider();
 
-  const getBalanceData = useCallback(async () => {
-    setLoading(true);
-    try {
-      if (!selectedWallet?.provider) return;
-      const balance = await getAccountBalance(
-        selectedWallet?.provider,
-        selectedAccount as string
-      );
-
-      const chain = chainData.find(
-        (item) => BigInt(item.chainId) === BigInt(chainId)
-      );
-
-      setAccountBalance(balance);
-      setCurrentChain(chain);
-    } catch (error) {
-      console.error(error);
-      processErrorMessage(error);
-    } finally {
-      setLoading(false);
-    }
-  }, [
-    chainId,
-    getAccountBalance,
-    processErrorMessage,
-    selectedAccount,
-    selectedWallet?.provider,
-  ]);
-
   useEffect(() => {
-    getBalanceData();
-  }, [getBalanceData, selectedAccount, selectedWallet?.provider]);
+    const chain = chainData.find(
+      (item) => BigInt(item.chainId) === BigInt(chainId)
+    );
+
+    setAccountBalance(currentBalance);
+    setCurrentChain(chain);
+  }, [chainId, currentBalance, selectedAccount]);
```

Comment:
1. `selectedAccount` is not being used in the useEffect hook, so it should be removed from the dependency array
2. Why not use currentBalance directly instead of creating a new state?
3. If we do need to use a new state for some reason, we should avoid combining different logics in a useEffect; please keep them separate.

Reply:
Sorry it's my fault, I used to fetch account balance inside this component before moving it to the context and I forgot to remove the state, I will update immediately.